osc/osc.go

```diff
@@ -926,79 +928,55 @@ func writeBlob(data []byte, buf *bytes.Buffer) (int, error) {
 	}
 
 	// Write the data
-	if _, err := buf.Write(data); err != nil {
-		return 0, nil
-	}
+	n, _ := buf.Write(data)
 
 	// Add padding bytes if necessary
-	numPadBytes := padBytesNeeded(len(data))
-	if numPadBytes > 0 {
-		padBytes := make([]byte, numPadBytes)
-		n, err := buf.Write(padBytes)
-		if err != nil {
-			return 0, err
-		}
-		numPadBytes = n
-	}
+	numPadBytes := padBytesNeeded(n)
+	buf.Write(padBytes[:numPadBytes])
```

Comment:
I figured that instead of creating a new slice with the right length or using a loop and `WriteByte` it just made more sense to pre-allocate a `[]byte` with 4 `0`'s and only write the amount required..